R/Spectra.R

```diff
@@ -1887,11 +1927,27 @@ setReplaceMethod("smoothed", "Spectra", function(object, value) {
 #' @importMethodsFrom ProtGenerics spectraData
 #'
 #' @exportMethod spectraData
-setMethod("spectraData", "Spectra", function(object,
-                                             columns = spectraVariables(object))
-{
-    spectraData(object@backend, columns = columns)
-})
+setMethod(
+    "spectraData", "Spectra",
+    function(object, columns = spectraVariables(object)) {
+        if (length(object@processingQueue) &&
```

Comment:
Just a suggestion, but I would find it useful to have some comments that describe the different situations here.

Reply:
you mean within the code?

Reply:
I added comments in the code. Basically, if the processing queue is empty or the user does **not** request peak variables with `columns` the `spectraData` method from the backend is called. Otherwise we need to execute the processing queue (calling `peaksData` on the `Spectra` object) to ensure that e.g. peaks are filtered etc and add the individual peaks variables to the returned `DataFrame`.